sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1519,12 +1519,7 @@ class Analyzer(override val catalogManager: CatalogManager)
         }
       // If the script transformation input contains Stars, expand it.
       case t: ScriptTransformation if containsStar(t.input) =>
-        t.copy(
-          input = t.input.flatMap {
-            case s: Star => s.expand(t.child, resolver)
-            case o => o :: Nil
-          }
-        )
+        t.copy(input = t.child.output)
```

Comment:
Could you add tests for `UnresolvedRegex` (`spark.sql.parser.quotedRegexColumnNames`=true case)?

Reply:
Done， can be supported.